packages/nodes-base/nodes/Google/Drive/GoogleDrive.node.ts

```diff
@@ -2526,56 +2524,100 @@ export class GoogleDrive implements INodeType {
+								const metadata = await this.helpers.getBinaryMetadata(binary.id);
+								contentLength = metadata.fileSize;
+								originalFilename = metadata.fileName;
+								if (metadata.mimeType) mimeType = binary.mimeType;
+							} else {
+								fileContent = Buffer.from(binary.data, BINARY_ENCODING);
+								contentLength = fileContent.length;
+								originalFilename = binary.fileName;
+								mimeType = binary.mimeType;
 							}
-
-							body = await this.helpers.getBinaryDataBuffer(i, propertyNameUpload);
 						} else {
 							// Is text file
-							body = Buffer.from(this.getNodeParameter('fileContent', i) as string, 'utf8');
+							fileContent = Buffer.from(this.getNodeParameter('fileContent', i) as string, 'utf8');
+							contentLength = fileContent.byteLength;
 						}
 
 						const name = this.getNodeParameter('name', i) as string;
 						const parents = this.getNodeParameter('parents', i) as string[];
 
-						let qs: IDataObject = {
-							fields: queryFields,
-							uploadType: 'media',
-						};
-
-						const requestOptions = {
-							headers: {
-								'Content-Type': mimeType,
-								'Content-Length': body.byteLength,
-							},
-							encoding: null,
-							json: false,
-						};
-
-						let response = await googleApiRequest.call(
-							this,
-							'POST',
-							'/upload/drive/v3/files',
-							body,
-							qs,
-							undefined,
-							requestOptions,
-						);
+						let uploadId;
+						if (Buffer.isBuffer(fileContent)) {
+							const response = await googleApiRequest.call(
+								this,
+								'POST',
+								'/upload/drive/v3/files',
+								fileContent,
+								{
+									fields: queryFields,
+									uploadType: 'media',
+								},
+								undefined,
+								{
+									headers: {
+										'Content-Type': mimeType,
+										'Content-Length': contentLength,
+									},
+									encoding: null,
+									json: false,
+								},
+							);
+							uploadId = JSON.parse(response).id;
+						} else {
+							const resumableUpload = await googleApiRequest.call(
+								this,
+								'POST',
+								'/upload/drive/v3/files',
+								undefined,
+								{ uploadType: 'resumable' },
+								undefined,
+								{
+									resolveWithFullResponse: true,
+								},
+							);
+							const uploadUrl = resumableUpload.headers.location;
+
+							let offset = 0;
+							for await (const chunk of fileContent) {
+								const nextOffset = offset + chunk.length;
+								try {
+									const response = await this.helpers.httpRequest({
+										method: 'PUT',
+										url: uploadUrl,
+										headers: {
+											'Content-Length': chunk.length,
+											'Content-Range': `bytes ${offset}-${nextOffset - 1}/${contentLength}`,
+										},
+										body: chunk,
+									});
+									uploadId = response.id;
+								} catch (error) {
+									if (error.response?.status !== 308) throw error;
+								}
+								console.log(
```

Comment:
we should remove this. but, we should look into passing this info back to the UI to be able to show upload progress.

Reply:
I think it should be removed so we don't flood the logs.